go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -71,6 +72,33 @@ func validateSparkJob(sparkJob *plugins.SparkJob) error {
 	return nil
 }
 
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func populateSparkConfig(sparkConfig, userSparkConfig map[string]string) error {
+	for k, v := range userSparkConfig {
+		if k == "spark.executor.instances" && sparkConfig["spark.executor.instances"] != "" {
```

Comment:
The current behavior of the config value is to be the default value if the users don't provide one. This will change its behavior as the max.

 Let's add a a separate section for limits ?

Reply:
Updated with separate section.